MoreLinq.NoConflictGenerator/Program.cs

```diff
@@ -0,0 +1,455 @@
+                            select CreateTypeKey(p.Type,
+                                                 n => typeParameterAbbreviationByName != null
+                                                   && typeParameterAbbreviationByName.TryGetValue(n, out var a) ? a : null),
+                    }
+                select m;
+
+        var q =
+            from e in qq.Select((m, i) => (SourceOrder: i + 1, Method: m))
+            orderby
+                e.Method.Name,
+                e.Method.TypeParameterCount,
+                e.Method.ParameterCount,
+                new TupleTypeKey(ImmutableList.CreateRange(e.Method.SortableParameterTypes))
+            select new
+            {
+                e.Method,
+                e.SourceOrder,
+            };
+
+            q = q.ToArray();
+
+            if (debug)
+            {
+                foreach (var m in
+                    from e in q
+                    let m = e.Method
+                    select new
+                    {
+                        m.Name,
+
+                        SourceOrder = e.SourceOrder.ToString("000", CultureInfo.InvariantCulture),
+
+                        TypeParameters =
+                            m.TypeParameterCount == 0
+                            ? string.Empty
+                            : "<" + string.Join(", ", from a in m.TypeParameterAbbreviationByName
+                                                      select a.Value) + ">",
+                        Parameters =
+                            "(" + string.Join(", ", m.SortableParameterTypes) + ")",
+
+                        Abbreviations =
+                            m.TypeParameterCount == 0
+                            ? string.Empty
+                            : " where " + string.Join(", ", from a in m.TypeParameterAbbreviationByName
+                                                            select a.Value + " = " + a.Key),
+                    }
+                    into e
+                    select e.SourceOrder + ": "
+                         + e.Name + e.TypeParameters + e.Parameters + e.Abbreviations)
+                {
+                    Console.Error.WriteLine(m);
+                }
+            }
+
+            var indent = new string('\x20', 4);
+            var indent2 = indent + indent;
+            var indent3 = indent2 + indent;
+
+            var baseImports = new []
+            {
+                "System",
+                "System.CodeDom.Compiler",
+                "System.Collections.Generic",
+            };
+
+            var imports =
+                from ns in baseImports.Concat(usings)
+                select indent + $"using {ns};";
+
+            var classes =
+                from md in q
+                select md.Method.Syntax into md
+                group md by (string) md.Identifier.Value into g
+                select new
+                {
+                    Name = g.Key,
+                    Overloads =
```

Comment:
I think this subquery could be (partly) extracted into a method:

```chsarp
MethodDeclarationSyntax CreateForwardingMethodDeclaration(MethodDeclarationSyntax originalMethod) =>
    MethodDeclaration(originalMethod.ReturnType, originalMethod.Identifier)
        .WithAttributeLists(originalMethod.AttributeLists)
        .WithModifiers(originalMethod.Modifiers)
        .WithTypeParameterList(originalMethod.TypeParameterList)
        .WithConstraintClauses(originalMethod.ConstraintClauses)
        .WithParameterList(originalMethod.ParameterList)
        .WithExpressionBody(
            ArrowExpressionClause(
                InvocationExpression(
                    MemberAccessExpression(
                        SyntaxKind.SimpleMemberAccessExpression,
                        IdentifierName("MoreEnumerable"),
                        IdentifierName(originalMethod.Identifier)),
                    ArgumentList(SeparatedList(
                        from p in originalMethod.ParameterList.Parameters
                        select Argument(IdentifierName(p.Identifier)))))
                    .WithLeadingTrivia(Space))
                .WithLeadingTrivia(Whitespace(indent3)))
        .WithSemicolonToken(ParseToken(";").WithTrailingTrivia(LineFeed));
var classes =
    from md in q
    select md.Method.Syntax into md
    group md by (string) md.Identifier.Value into g
    select new
    {
        Name = g.Key,
        Overloads = g.Select(CreateForwardingMethodDeclaration),
    }
    // ....
```

Reply:
[Same sentiment as earlier](https://github.com/morelinq/MoreLINQ/pull/235#discussion_r177911201).